otelcol/unmarshaler.go

```diff
@@ -32,6 +27,10 @@ type configSettings struct {
 // unmarshal the configSettings from a confmap.Conf.
 // After the config is unmarshalled, `Validate()` must be called to validate.
 func unmarshal(v *confmap.Conf, factories Factories) (*configSettings, error) {
+
+	telFactory := telemetry.NewFactory()
```

Comment:
I recognize this works, but it feels kinda arbitrary to create a new factory here.  Would it be better to create a new telemetry factory at the same level we create the `Factories` and then pass it around?

Reply:
I want to do do this indeed, but I want to keep public API changes in service and friends for another PR. Would that be okay with you? Or do you want to do it all in one go?

Reply:
Ya thats ok